src/adagrams.js

```diff
@@ -1,8 +1,95 @@
 const Adagrams = {
   drawLetters() {
-    // Implement this method for wave 1
+    let letterPool = [];
+    const letterChoices = {
+      A: 9, B: 2, C: 2, D: 4, E: 12, F: 2, G: 3, H: 2,
+      I: 9, J: 1, K: 1, L: 4, M: 2, N: 6, O: 8, P: 2, Q: 1, R: 6,
+      S: 4, T: 6, U: 4, V: 2, W: 2, X: 1, Y: 2, Z: 1,
+    };
+
+    for (const letter in letterChoices) {
+      if (letterChoices !== undefined) {
+        const charCount = letterChoices[letter];
+        const repeatSplit = letter.repeat(charCount).split('');
+        letterPool = letterPool.concat(repeatSplit);
+      }
+    }
+
+    const userHand = [];
+    // used this solution to get random sampling from array https://stackoverflow.com/questions/4550505/getting-a-random-value-from-a-javascript-array
+    for (let i = 0; i < 10; i++) {
+      const sample = letterPool[Math.floor(Math.random() * letterPool.length)];
+      userHand.push(sample);
+    };
+    return userHand;
+  },
+
+  usesAvailableLetters(input, lettersInHand) {
+    const lettersHash = {};
+
+    lettersInHand.forEach((letter) => lettersHash[letter]?
+    lettersHash[letter] = lettersHash[letter] += 1 : lettersHash[letter] = 1);
```

Comment:
This isn't reading very well; it's probably important in this case to have all of the curly brackets. Consider:

```javascript
lettersInHand.forEach((letter) => {
  lettersHash[letter] = lettersHash[letter] ? lettersHash[letter] += 1 : lettersHash[letter] = 1;
});
```